main.go

```diff
@@ -92,10 +92,11 @@ func mainError() error {
 			Logger:        logger,
 			ServerFactory: serverFactory,
 
-			Description: description,
-			GitCommit:   gitCommit,
-			Name:        name,
-			Source:      source,
+			Description:    description,
+			GitCommit:      gitCommit,
+			Name:           name,
+			Source:         source,
+			VersionBundles: service.NewVersionBundles(),
```

Comment:
What is this used for here? Will there be `version` command as suggested in PR title?

Reply:
Yes this is pretty magical at first glance. `command.Command` adds two commands to cobra: `daemon` and `version`. `daemon` uses ServerFactory and `version` rest of the values in the Config. I think it would be good to be more explicit and separate those at expense of a few lines of code. I even was tempted to do so with my latest refactoring but it was enough changes already.


See that:

```
$ go run main.go version
description: The azure-operator manages Kubernetes clusters on Azure.
gitCommit: n/a
name: azure-operator
source: https://github.com/giantswarm/azure-operator
goVersion: go1.10
os: darwin
arch: amd64
versionBundles:
- changelogs:
  - component: kubernetes
    description: Enabled encryption at rest.
    kind: added
  - component: kubernetes
    description: Updated to version 1.9.2.
    kind: changed
  - component: kubernetes
    description: Switched vanilla (previously coreos) hyperkube image.
    kind: changed
  - component: calico
    description: Updated to version 3.0.1.
    kind: changed
  - component: containerlinux
    description: Updated to version 1632.3.0.
    kind: changed
  - component: docker
    description: Updated to version 17.09.0.
    kind: changed
  - component: calico
    description: Removed calico-ipip-pinger.
    kind: removed
  - component: calico
    description: Removed calico-node-controller.
    kind: removed
  - component: cloudconfig
    description: Replace systemd units for Kubernetes components with self-hosted
      pods.
    kind: changed
  - component: azure-operator
    description: Change default etcd data dir to /var/lib/etcd.
    kind: changed
  components:
  - name: calico
    version: 3.0.1
  - name: containerlinux
    version: 1632.3.0
  - name: docker
    version: 17.09.0
  - name: etcd
    version: 3.2.7
  - name: coredns
    version: 1.0.5
  - name: kubernetes
    version: 1.9.2
  - name: nginx-ingress-controller
    version: 0.10.2
  dependencies: []
  deprecated: false
  name: azure-operator
  time: !!timestamp 2018-01-07T08:35:00Z
  version: 0.1.0
  wip: true
```